lib/iris/analysis/maths.py

```diff
@@ -342,13 +374,12 @@ def divide(cube, other, dim=None, in_place=False):
     other_unit = getattr(other, 'units', '1')
     new_unit = cube.units / other_unit
     if in_place:
-        op = operator.itruediv
-        # Input `other` may not have a `dtype` attribute.
-        other_dtype_kind = other.dtype.kind if hasattr(other, 'dtype') else 'f'
-        if 'i' in [cube.dtype.kind, other_dtype_kind]:
-            # NumPy ufunc `true_divide` cannot coerce output to int.
-            aemsg = 'Cannot perform inplace division of integer data.'
+        if cube.dtype.kind == 'i':
+            # Cannot coerce float result from inplace division back to int.
+            aemsg = ('Cannot perform inplace division of cube {!r} '
+                     'with integer data.')
```

Comment:
I'll investigate... but it will probably need a `kwarg` to `_inplace_common`, which may just move the bad stuff around. The reason I didn't use it for divide is because divide has a quite different processing requirement and there wasn't a whole lot of code duplication...